hermes/src/store/storage.rs

```diff
@@ -27,8 +19,15 @@ pub enum Key {
 /// data to abstract the details of the update data, and so each update data is stored
 /// under a separate key. The caller is responsible for specifying the right
 /// key for the update data they wish to access.
-pub trait Storage: Sync + Send {
-    fn insert(&self, key: Key, time: UnixTimestamp, value: StorageData) -> Result<()>;
-    fn get(&self, key: Key, request_time: RequestTime) -> Result<Option<StorageData>>;
-    fn keys(&self) -> Vec<Key>;
+pub trait Storage: Send + Sync {
+    fn store_message_states(&self, message_states: Vec<MessageState>) -> Result<()>;
```

Comment:
I think the trait should be more like put(key, value) than storing an object that internally has a key in it. These two function signatures are confusing -- it's hard for the reader to tell how to retrieve something they inserted.

Reply:
after reading through the rest of the code, I think this abstraction should be called something like `TemporalCache<K, V>` and the function signatures should be `put(key: K, timestamp, value: V)` and `get(key: K, request_time): Option<V>`.

Reply:
yeah i agree with your suggestion.